src/cli/run.rs

```diff
@@ -85,14 +91,106 @@ pub async fn run<P: AsRef<Path>>(
     )?;
 
     if task_outcome == TaskOutcome::Failure {
+        stop_notify.notify_one();
+        let _ = inquiry_handle.await;
         return Ok(());
     }
 
     squash_merge_branch(path, &base_branch, &fork_branch)?;
-
+    stop_notify.notify_one();
+    let _ = inquiry_handle.await;
     Ok(())
 }
 
+async fn handle_inquiries(agent_base_url: String, agent_api_key: String, stop_notify: Arc<Notify>) {
+    let client = reqwest::Client::new();
+
+    loop {
+        // Check stop signal
+        let stop_future = stop_notify.notified();
+        let sleep_future = tokio::time::sleep(std::time::Duration::from_secs(1));
+        tokio::select! {
+            _ = stop_future => {
+                println!("[handle_inquiries] Received stop_notify. Shutting down handler!");
+                break;
+            }
+            _ = sleep_future => {
+                let url = format!("{agent_base_url}/api/agent/inquiry_request");
+                let resp_result = client.get(&url).bearer_auth(&agent_api_key).send().await;
+                match resp_result {
+                    Ok(resp) => {
+                        let text = resp.text().await.unwrap_or_else(|e| {
+                            println!("[handle_inquiries] ERROR reading response text: {e}");
+                            "".to_owned()
+                        });
+                        let question = text.trim().to_string();
+
+                        if !question.is_empty() {
+                            // Fancy ORCA inquiry request
+                            let width = term_size::dimensions().map(|(w, _)| w).unwrap_or(80);
+
+                            let banner = "🐋=== ORCA WANTS INPUT ===🐋";
```

Comment:
The CLI is intended to work for agents other than ORCA, so let's replace any reference of ORCA by referring to *the agent* instead.